distributed/protocol/serialize.py

```diff
@@ -193,9 +197,58 @@ def register_serialization_family(name, dumps, loads):
 register_serialization_family("error", None, serialization_error_loads)
 
 
+def infer_if_recurse_to_serialize_list(x: list) -> bool:
+    """
+    Function sets iterate_collection variable in serialize for
+    lists.  If it returns False, the list will be serialized
+    using Dask Dispatch.  It it returns True, the list will
+    be serialized recursively with pickle
+    """
+    try:
+        _ = list(set(x))
```

Comment:
What is this for? This will iterate twice over the list and hash every element. That might be quite an expensive operation

Reply:
Thanks @fjetter.  The original thought was to be sure everything in the list is hashable or force recursive serialization.  In retrospect, we can leverage the `first_val` dtype check to validate this.  I'll update.